### ISSUE TITLE
Add option to ignore files

### ISSUE THREAD

**Issue by SetTrend** [DavidAnson/markdownlint#159]
Please add a configuration opportunity to exclude particular files from the linter.

Either allow globbing for the `config.files` option:

```
files: ["**/[a-cA-C]*.md"]
```

Or add a new configuration option, explicitly excluding files (again, with globbin):

```
exclude: ["**/__*.md"]
```

**Comment by DavidAnson** [comment]
Can you explain this request a bit more, please? As the caller, you control the contents of the files array - why not filter out the files you want before passing in the files array? You could use glob, or you could use any other technique. The current implementation allows complete flexibility to the caller.

(Aside: markdownlint-cli supports glob and the ability to exclude by glob - perhaps it is a good reference example?)

**Comment by SetTrend** [comment]
Thanks, David,

I'm participating in the [documentation repository of WebPack](https://github.com/webpack/webpack.js.org/issues/2632#issuecomment-446844383). Doing so, I wrote a script that concatenates their single MD pages into comprehensive chapter MD pages, so their information can be viewd/printed in one single document.

That's all highly dynamic. No static maintenance should be required to utilize this feature. So I need wildcards to be able to exclude the generated files (in this particular project their filenames all start with an underscore) from the linter, since, of cource, chances are that headings are being duplicated in the conglomerate.

**Comment by SetTrend** [comment]
I just went through the scripts in the `package.json` file of the project and noticed they are using `markdown-cli`. I will see if I can use globbing there, as you suggested. Thanks for the hint! As soon as I success I will close this issue then.

**Comment by SetTrend** [comment]
Works like a charm. Thanks!

**Comment by elgalu** [comment]
Also setting `--ignore` in a pre-commit works like a charm, e.g.

```yaml
  - repo: https://github.com/igorshubovych/markdownlint-cli
    rev: "596386e6a7da92547c4fc6b94fce59bcbc8290ee" # frozen: v0.27.1
    hooks:
      - id: markdownlint
        args: ["--ignore", "towncrier-template.md"]
```

**Comment by BenjaminBrienen** [comment]
@DavidAnson I would like to include/exclude files in the markdownlint config file because controlling it via the command line flags means that despite putting the ignore on the CI workflow, it still gives warnings in the language server or local usages of the CLI. I want all contributors and all contexts to share this requirement - in this case, a generated file should not be analyzed by the linter.
